api/src/game_handlers.rs

```diff
@@ -0,0 +1,655 @@
+}
+/**
+ *  Tests for the Web API.  The actual logic of the game is already tested in the unit tests for that part of the project
+ *
+ *  these tests test the "shape" of the Web API - making sure that serialization/deserialization works (by simply using it
+ *  in the tests) and that the response we get back are non-empty or have other reasonable values.
+ *
+ *  reproducibility isn't something the game uses, but is verified here to enforce the correctness of the repeat URL
+ *
+ */
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::game_handlers;
+    use actix_web::{test, web, App};
+    use cribbage_library::scoring::CombinationName;
+
+    macro_rules! get_repeat_url {
+        ($url: expr) => {{
+            let index = $url.find("/").unwrap() + 1;
+            let repeat_url = $url
+                .chars()
+                .skip(index - 1) // we want the '/' to start the string
+                .take($url.len() - index + 1)
+                .collect::<String>();
+
+            println!("repeat url: {}", repeat_url);
+            repeat_url
+        }};
+    }
+
+    macro_rules! vec_contains_card {
+        ($vec: expr, $card: expr) => {{
+            let mut found = false;
+            for c in $vec.iter() {
+                if c.cardName == $card {
+                    found = true;
+                    break;
+                }
+            }
+            found
+        }};
+    }
+    #[allow(unused_macros)]
+    macro_rules! vec_contains_score {
+        ($vec: expr, $name: expr) => {{
+            let mut found = false;
+            for score in $vec.iter() {
+                if score.ScoreName == $name {
+                    found = true;
+                    break;
+                }
+            }
+            found
+        }};
+    }
+
+    /**
+     *  this macro takes in the service to call, the orignal response, and the repeat URL
+     *  it then calls the service again with the repeat URL and verifies that the return
+     *  values are identical.
+     */
+    macro_rules! test_repeatability {
+        ($service:expr, $original_response:expr, $un_parsed_repeat_url:expr) => {{
+            let repeat_url = get_repeat_url!($un_parsed_repeat_url);
+            let req = test::TestRequest::get().uri(&repeat_url).to_request();
+            let response = test::read_response(&mut $service, req).await;
+            let repeat_json = response;
+            let json_original = serde_json::to_string(&$original_response).unwrap();
+            assert_eq!(actix_web::web::Bytes::from(repeat_json), json_original);
+        }};
+    }
+
+    //
+    //  this test gets the cut cards and the parses out the repeat URL to call cutcards again
+    //  it verifies that it gets the same results back.
+    //
+    //  this also verifies that the json serialize/deserialize of the CutCardResponse works correctly
+    //
+    #[actix_rt::test]
+    async fn cut_cards() {
+        //
```

Comment:
A blank line would look better here as a break than the comment start.

Reply:
ah...this is done by the code formatter.